syntax = "proto3";

package fixbug;

option cc_generic_services = true;

message ErrorMsg
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

message LoginRequest
{
    bytes name = 1;    
    bytes pwd = 2;   
}

message LoginResponse
{
    ErrorMsg error_msg = 1;
    bool success = 2;
}

message SumRequest
{
    repeated int32 nums = 1;
}

message SumResponse
{
    ErrorMsg error_msg = 1;
    int32 res = 2;
}

service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Sum(SumRequest) returns(SumResponse);
}